backend/modules/query_engines/default_engine.py

```diff
@@ -0,0 +1,70 @@
+from fastapi import HTTPException
+from langchain.chains import RetrievalQA
+from langchain.chains.question_answering import load_qa_chain
+from langchain.prompts import PromptTemplate
+from langchain.schema.vectorstore import VectorStoreRetriever
+
+from backend.modules.query_engines.base import BaseQueryEngine
+from backend.utils.base import DefaultQueryInput
+from backend.utils.decorator import Post, QueryEngine
+from backend.utils.logger import logger
+
+
+@QueryEngine("/default")
```

Comment:
The decorator can help you register.

Reply:
kinda chicken and egg
I think all module classes will have to be imported in some place in __init__.py for the decorator to even execute
or some code at boot time will have to scan directories and find all classes

Reply:
Yes, we ultimately need to migrate to library based route